src/functions/uploadToRSIS/application/data-mapper/cat-cpc-mapper.ts

```diff
@@ -0,0 +1,111 @@
+import { ResultUpload } from '../result-client';
+import { DataField, FormType } from '../../domain/mi-export-data';
+import { TestData, TestResultCatCPCSchema } from '@dvsa/mes-test-schema/categories/CPC';
+import { addIfSet, field, mandatory, optional, optionalBoolean } from './data-mapper';
+import { formatApplicationReference } from '@dvsa/mes-microservice-common/domain/tars';
+import moment = require('moment');
+import { formatDateOfBirth, formatResult, formatLanguage } from './common-mapper';
+
+export const mapCatCPCData = (result: ResultUpload): DataField[] => {
+  const testResult = result.testResult as TestResultCatCPCSchema;
+  const t: TestData = result.testResult.testData as TestData;
+  const testDateTime = moment(testResult.journalData.testSlotAttributes.start, 'YYYY-MM-DDTHH:mm:ss');
+  const candidateDOB = moment(formatDateOfBirth(result)).format('YYMMDD');
+
+  const m: DataField[] = [
+    field('ACTIVITY_CODE', Number(testResult.activityCode)),
+    // unused - ADI_PRN
+    field('APP_REF_NO', formatApplicationReference(testResult.journalData.applicationReference)),
+    field('C', optionalBoolean(testResult, 'changeMarker')),
+    field('CANDIDATE_SURNAME', mandatory(testResult, 'journalData.candidate.candidateName.lastName')),
+    field('CAT_TYPE', formatCPCTestCategory(testResult)),
+    // unused - DATA_VALIDATION_FLAGS
+    field('DATE_OF_TEST', testDateTime.format('YYMMDD')),
+    field('DEBRIEF_GIVEN', testResult.activityCode === '51' ? 0 : 1),
+    field('DRIVER_NUMBER', mandatory(testResult, 'journalData.candidate.driverNumber')),
+    field('DTC_AUTHORITY_CODE', testResult.journalData.testCentre.costCode),
+    // unused - EXAMINER_FORENAMES
+    field('EXAMINER_PERSON_ID', testResult.journalData.examiner.staffNumber),
+    // unused - EXAMINER_SURNAME
+    field('FORM_TYPE', FormType.CPC),
+    // unused - IMAGE_REFERENCE
+    // unused - INSTRUCTOR_CERT
+    field('INT', optionalBoolean(testResult, 'accompaniment.interpreter')),
+    // unused - REC_NO
+    // unused - REC_TYPE
+    // unused - REG_NO
+    // SEC1
+    field('SEC1_MARK1', optionalBoolean(t, 'question1.answer1.selected')),
+    field('SEC1_MARK2', optionalBoolean(t, 'question1.answer2.selected')),
+    field('SEC1_MARK3', optionalBoolean(t, 'question1.answer3.selected')),
+    field('SEC1_MARK4', optionalBoolean(t, 'question1.answer4.selected')),
+    field('SEC1_PERCENT_SCORE', optional(t, 'question1.score', 0)),
+    field('SEC1_Q_NO', optional(t, 'question1.questionCode', '')),
```

Comment:
is an empty string the correct default value instead of a null? same applies for all other questionCode mappings